pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -546,6 +553,31 @@ File getSegmentDataDir(String segmentName) {
     return new File(_indexDir, segmentName);
   }
 
+  @VisibleForTesting
+  File getSegmentDataDir(String segmentName, String segmentTier, IndexLoadingConfig indexLoadingConfig) {
+    if (segmentTier == null) {
+      return getSegmentDataDir(segmentName);
+    }
+    try {
+      String tierDataDir = TierConfigUtils
+          .getDataDirFromTierConfig(_tableNameWithType, segmentTier, indexLoadingConfig.getTableConfig());
+      File tierTableDataDir = new File(tierDataDir, _tableNameWithType);
+      return new File(tierTableDataDir, segmentName);
+    } catch (Exception e) {
+      LOGGER.warn("Failed to get dataDir for segment: {} of table: {} on tier: {} due to error: {}", segmentName,
+          _tableNameWithType, segmentTier, e.getMessage());
+      return getSegmentDataDir(segmentName);
+    }
+  }
+
+  private String getSegmentCurrentTier(String segmentName) {
```

Comment:
(minor) Annotate the return as `Nullable`

Reply:
kk, I thought those annotations are for public methods.